docs/ldml/tr35-numbers.md

```diff
@@ -1423,12 +1423,22 @@ The above description describes the expected output. Internally, the implementat
 
 Two semantically equivalent tokens can be *collapsed* if they appear at the start of both values or the end of both values. However, the implementation may choose different levels of aggressiveness with regard to collapsing tokens. The currently recommended heuristic is:
 
-1. Only collapse semantically equivalent *unit of measurement* tokens. This is to avoid ambiguous strings such as "3–5K" (could represent 3–5000 or 3000–5000).
+1. Never collapse scientific or compact notation. This is to avoid producing ambiguous strings such as "3–5M" (could represent 3–5,000,000 or 3,000,000–5,000,000).
 1. Only collapse if the tokens are more than one code point in length. This is to increase clarity of strings such as "$3–$5".
 
-These heuristics may be refined in the future.
+To perform the collapse:
+
+1. Remove the token that is closest to the range separator.
+That is, for a prefix element, remove from the end value, and for a suffix element remove it from the start value:
+    * USD 2 – USD 5 ⇒ USD 2 – 5
+    * 2M EUR – 5M EUR ⇒  2 M – 5M EUR
+    * 2 km – 5 km ⇒ 2 – 5 km
+    * 2M ft – 5M ft ⇒ 2M – 5M ft
+2. In bidi contexts, rule #1 is applied **visually**. For example, if a range from 2 km to 5 km would be presented visually as "_mk 5 – mk 2_", the collapsed form would be "_mk 5 – 2_". (The _mk_ is a stand-in for the native representation.)
```

Comment:
Question about the new bidi clause. The example you gave should be fine for both visual and logical order, right? In the ICU implementation we definitely only do this in logical order (because visual order would require pulling in the bidi code, which we don't do).

Reply:
Revised, with the bidi note restructured.